src/scanner/index.ts

```diff
@@ -35,6 +33,38 @@ export async function sendDeleteWorkloadRequest(workloadName: string, localWorkl
   await deleteWorkload(deletePayload);
 }
 
+export function getUniqueImages(workloadMetadata: IWorkload[]): IScanImage[] {
+  const uniqueImages: { [key: string]: IScanImage } = workloadMetadata.reduce((accum, meta) => {
+    logger.info({
+      workloadName: workloadMetadata[0].name,
+      name: meta.imageName,
+      id: meta.imageId
+    }, 'image metadata');
+    // example: For DCR "redis:latest"
+    // example: For GCR "gcr.io/test-dummy/redis:latest"
+    // example: For ECR "291964488713.dkr.ecr.us-east-2.amazonaws.com/snyk/redis:latest"
+    // meta.imageName can be different depends on CR
+    const { imageName } = getImageParts(meta.imageName);
+    // meta.imageId can be different depends on CR
+    // example: For DCR "docker.io/library/redis@sha256:8e9f8546050da8aae393a41d65ad37166b4f0d8131d627a520c0f0451742e9d6"
+    // example: For GCR "sha256:8e9f8546050da8aae393a41d65ad37166b4f0d8131d627a520c0f0451742e9d6"
+    // example: For ECR "sha256:8e9f8546050da8aae393a41d65ad37166b4f0d8131d627a520c0f0451742e9d6"
+    let digest: string | undefined = undefined;
+    if (meta.imageId.lastIndexOf('@') > -1 || meta.imageId.startsWith('sha')) {
```

Comment:
only scan by digest if we sure it has it

Reply:
Shouldn't we try to remove all URI scheme prefixes first before looking for the SHA? So that we can handle things like `docker-pullable://`?

Reply:
Prefer doing this out of data from logger. Cos we still kinda blind what `imageId` can be in all variations.

Reply:
We'll test it in the environments we have? We have GKE, EKS, OpenShift but nothing on AKS or any other managed service 🤔 But I guess it's a good start